intermediate-python-debugging/vs-code-debugging.md

```diff
@@ -0,0 +1,190 @@
+# Debugging With VS Code
+
+Python is widely used with VS Code and so VS Code's developers have put the time in to integrate the Python debugger into the standard VS Code plugin for Python.
+
+In this lesson we will learn to connect VS Code and the Python debugger to help us fix errors in our code.
+
+## Learning Goals
+
+By the end of this lesson we should be able to...
+
+- Set up VS Code to run the Python debugger
+- Run the Python debugger in VS Code to step through a program
+- Use breakpoints, and watches to identify errors in code
```

Comment:
Another suggestion I have is to convert this into a "Goals and Structure" section. This will establish what the lesson is trying to do and how. This is useful to me because this lesson reads way more like a directed/guided exercise with steps in order (with a lot of "you do this now, observe this now, answer this now"), rather than disseminating knowledge.

```suggestion
## Goal and Structure

In order to use a debugger in VS Code while working on Python projects, we need to set up our tools and then learn how to navigate and use them.

This lesson is broken up into the following sections:

1. Setting up VS Code to use and run the Python debugger
2. Using a small example Python program to learn how to and practice:
    - Adding and removing breakpoints
    - Running the debugger and watching the program pause
    - Observing variable state at a breakpoint
    - Creating and using a watch
    - Resuming, stepping over, stepping into, step out, restart, and stop
```